dev/com.ibm.ws.microprofile.config.1.1_fat/fat/src/com/ibm/ws/microprofile/config/fat/tests/visibility/war/VisibilityTestServlet.java

```diff
@@ -0,0 +1,282 @@
+        assertContains(config, EJB_JAR_CONFIG_PROPERTY, EJB_JAR_VALUE);
+        assertContains(config, EAR_LIB_CONFIG_PROPERTY, EAR_LIB_VALUE);
+        assertContains(config, WAR_CONFIG_PROPERTY, WAR_VALUE);
+
+        // Same bean obtained via EJB
+        config = testEjb.getEarLibBean().getConfig();
+
+        // Expect to see everything visible to the .war
+        // We get the bean via the EJB, but we call the bean ourselves, so it's the .war TCCL that's active
+        assertContains(config, EJB_JAR_CONFIG_PROPERTY, EJB_JAR_VALUE);
+        assertContains(config, EAR_LIB_CONFIG_PROPERTY, EAR_LIB_VALUE);
+        assertContains(config, WAR_CONFIG_PROPERTY, WAR_VALUE);
+
+        // Test with dependent-scoped ear lib jar
+        config = earLibDependentBean.getConfig();
+
+        // Expect same results as app-scoped bean
+        // The bean creation has no impact when calling getConfig
+        assertContains(config, EJB_JAR_CONFIG_PROPERTY, EJB_JAR_VALUE);
+        assertContains(config, EAR_LIB_CONFIG_PROPERTY, EAR_LIB_VALUE);
+        assertContains(config, WAR_CONFIG_PROPERTY, WAR_VALUE);
+
+        // Test with dependent-scoped ear lib jar obtained via EJB
+        config = testEjb.getEarLibDependentBean().getConfig();
+
+        // Expect to see everything visible to the .war
+        // We get the bean via the EJB, but we call the bean ourselves, so it's the .war TCCL that's active
+        assertContains(config, EJB_JAR_CONFIG_PROPERTY, EJB_JAR_VALUE);
+        assertContains(config, EAR_LIB_CONFIG_PROPERTY, EAR_LIB_VALUE);
+        assertContains(config, WAR_CONFIG_PROPERTY, WAR_VALUE);
+    }
+
+    @Test
+    public void testInjectedConfigFromWar() {
+        // Config injected into .war
+        Config config = injectedConfig;
+
+        // Expect to see properties from all locations visible on war classpath
+        assertContains(config, WAR_CONFIG_PROPERTY, WAR_VALUE);
+        assertContains(config, EJB_JAR_CONFIG_PROPERTY, EJB_JAR_VALUE);
+        assertContains(config, EAR_LIB_CONFIG_PROPERTY, EAR_LIB_VALUE);
+    }
+
+    @Test
+    public void testInjectedConfigFromEjbJar() {
+        // Config injected into EJB
+        Config config = testEjb.getInjectedConfig();
+
+        // Expect to see properties from all locations visible on EJB jar classpath
+        assertContains(config, EJB_JAR_CONFIG_PROPERTY, EJB_JAR_VALUE);
+        assertContains(config, EAR_LIB_CONFIG_PROPERTY, EAR_LIB_VALUE);
+
+        if (isMpConfig1x()) {
+            // In mpConfig 1.x, the injected Config bean is ApplicationScoped, and reports the same properties everywhere
+            // It gets initialized wherever it's first used (which is from the .war)
+            assertContains(config, WAR_CONFIG_PROPERTY, WAR_VALUE);
+        } else {
+            // In mpConfig 2.0 and later, the injected Config bean is Dependent-scoped and will take its values from the EJB module TCCL
```

Comment:
I admit the fact what properties the class can see is dependent on its CDI scope and not which archive it is in feels wrong. 

But of course that doesn't make the test wrong.